### PR TITLE
Refactor error messages generation

### PR DESCRIPTION
**Pull request by schaubl**
Related to:
- slipstream/SlipStreamUI#271
- slipstream/SlipStreamServerDeps#14

**XML**
*before*
```xml
<error code="404">Error: The server has not found anything matching the request URI (404 - Not Found)</error>
```
*after*
```xml
<error code="404" reason="Not Found" detail="The server has not found anything matching the request URI" specificationUri="http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.5">
The server has not found anything matching the request URI
</error>
```
**JSON**
*before*
```json
{
   "error": "404",
   "reason": "Not Found",
   "detail": "The server has not found anything matching the request URI"
}
```
*after*
```json
{
    "error": {
        "code": "404",
        "reason": "Not Found",
        "detail": "The server has not found anything matching the request URI",
        "specificationUri": "http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.5"
    }
}
```
**PLAIN TEXT**
*before*
```text
Error: The server has not found anything matching the request URI (404 - Not Found)
```
*after*
```txt
404 - Not Found
The server has not found anything matching the request URI
```

### CODE REVIEW

File: jar-service/src/main/java/com/sixsq/slipstream/application/CommonStatusService.java [hunk 1]
```diff
@@ -136,10 +158,20 @@ private Representation toJson(Status status) {
...
+			json.append(",\n        \"reason\": \"" + status.getReasonPhrase() + "\"");
+
+		if (status.getDescription() != null)
+			json.append(",\n        \"detail\": \"" + status.getDescription() + "\"");
+
+		if (status.getUri() != null)
+			json.append(",\n        \"specificationUri\": \"" + status.getUri() + "\"");
+
+		json.append("\n    }\n");
+		json.append("}");
```
Comment (konstan) [inline jar-service/src/main/java/com/sixsq/slipstream/application/CommonStatusService.java]: Could we use some Java native Map to json serialiser?

Comment (schaubl) [inline jar-service/src/main/java/com/sixsq/slipstream/application/CommonStatusService.java]: This is my "final" goal but currently there is not support for JSON is SS Server.
So my idea was to wait for the `json` branch to be merged and then use the same serializer to generate this error.

Comment (konstan) [inline jar-service/src/main/java/com/sixsq/slipstream/application/CommonStatusService.java]: OK.

**Comment by konstan** [comment]
Looks OK.  Merging.

**Pull request merged**